papaya-web/src/main/java/pl/edu/agh/papaya/notification/NotificationService.java

```diff
@@ -0,0 +1,172 @@
+package pl.edu.agh.papaya.notification;
+
+import com.google.common.collect.Iterables;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import pl.edu.agh.papaya.model.Notification;
+import pl.edu.agh.papaya.model.NotificationType;
+import pl.edu.agh.papaya.model.Sprint;
+import pl.edu.agh.papaya.model.SprintState;
+import pl.edu.agh.papaya.model.UserInProject;
+import pl.edu.agh.papaya.notification.message.EmailMessage;
+import pl.edu.agh.papaya.notification.message.Message;
+import pl.edu.agh.papaya.notification.message.WebhookMessage;
+import pl.edu.agh.papaya.repository.AvailabilityRepository;
+import pl.edu.agh.papaya.repository.NotificationRepository;
+import pl.edu.agh.papaya.service.sprint.SprintService;
+import pl.edu.agh.papaya.service.userinproject.UserInProjectService;
+
+@Service
+@SuppressWarnings({"PMD.BeanMembersShouldSerialize"})
+public final class NotificationService {
+
+    @Value("${notifications.frequency.minutes:1440}")
+    private Integer notificationFrequency;
+
+    private final SprintService sprintService;
+
+    private final UserInProjectService userInProjectService;
+
+    private final NotificationRepository notificationRepository;
+
+    private final AvailabilityRepository availabilityRepository;
+
+    private final JavaMailSender javaMailSender;
+
+    @Autowired
+    public NotificationService(SprintService sprintService, UserInProjectService userInProjectService,
```

Comment:
Tu nie wiem jak inaczej przez to `@Value("${notifications.frequency.minutes:1440}")`